### PR TITLE
vscode-extensions.discloud.discloud: init 2.21.2

### PR DESCRIPTION
**Pull request by diogomdp**
## Description of changes
Adding a new vscode extension: https://marketplace.visualstudio.com/items?itemName=discloud.discloud

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -1268,6 +1268,23 @@ let
...
+      discloud.discloud = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          publisher = "discloud";
+          name = "discloud";
+          version = "2.21.2";
+          hash = "sha256-es1WjKchxC2hIWOkIRuf5MqMjTYu6qcBgo8abCqTjFc=";
+        };
+        meta = {
+          changelog = "https://marketplace.visualstudio.com/items/discloud.discloud/changelog";
+          description = "This extension makes it easy to host and manage applications on Discloud through Visual Studio Code.";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:1280]:
```suggestion
          description = "A Visual Studio Code extension for hosting and managing applications on Discloud";
```

**Comment by superherointj** [comment]
Please squash commits for extension. Ideally, a single VSCode extension should be a single commit.

**Comment by astindev** [comment]
Thank you for the review and feedback @superherointj !

I've rebased and squashed my previous commits into a single commit as requested.
I apologize for any confusion...

Please let me know if I need to change anything else.

Thank you for your time.

**Comment by superherointj** [comment]
The code that adds yourself as a maintainer (globally) should be a separate commit.
The vscode extension code should be a single commit. (Which has nothing to do with you being a maintainer.)
This matters when managing the maintainer lifecycle.
By keeping it separate, it's possible to revert the commit that added yourself as maintainer later.

**Comment by astindev** [comment]
That makes more sense...

Thank you very much for your explanation.

**Review by superherointj — APPROVED** [review]

**Pull request merged**